cloud/aws/aws_env.h

```diff
@@ -255,6 +263,10 @@ class AwsEnv : public CloudEnvImpl {
     return cloud_env_options.dest_bucket.GetObjectPath();
   }
   
+  bool HasDestBucket() const override {
+    return cloud_env_options.dest_bucket.IsValid();
```

Comment:
Can you define these functions in `CloudEnv`, based on `GetDestBucketName` and `GetSrcBucketName`? That way you can avoid the additional virtual function calls.

Reply:
To do this right, the GetCloudEnvOptions() should be made non-virtual.  Is that what you want?   I think that should be done, but have not done it as part of this fix.

Reply:
Let's start with defining `HasDestBucket()` and friends as non-virtual (which we can do even without defining `GetCloudEnvOptions()` non-virtual) and then in a follow-up PR we can extend this to `GetCloudEnvOptions()`